api/tests/helpers/database.go

```
@@ -0,0 +1,116 @@
+		td.DB.Close()
+	}
+}
+
+// WithTransaction executes a function within a database transaction for isolation
+func (td *TestDatabase) WithTransaction(fn func(*sql.Tx)) {
+	tx, err := td.DB.Begin()
+	Expect(err).NotTo(HaveOccurred(), "Failed to begin transaction")
+	
+	defer func() {
+		if r := recover(); r != nil {
+			tx.Rollback()
```

golangci-lint (1.64.8)

Error return value of `tx.Rollback` is not checked

(errcheck)